### PR TITLE
[SL-UP] Remove two algo design for the wifi retry mechanism

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Description
Wi-Fi reconnect logi had two different methodologies : One during commissioning and one for standard operations. Having this double logic is unecessary and the fibonnaci like design is sufficient.

The commissioning design also required access to the BaseApplication which is in the app layer which  is not accessible from the platform layer.

There will be a follow up PR to implement a true fibonnaci retry mechanims since the current design just doubles the retry interval

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

**Comment by jmartinez-silabs** [comment]
Careful with the upstream. the restyle part bring changes to the retrieve script.

File: scripts/tools/silabs/retrieve_sl_commits.py [hunk 1]
```diff
@@ -41,7 +43,7 @@ def main():
     [PREFIX] commits:
     <full_commit_sha> -- <Commit_Title>
     """,
-    epilog= """
+                                     epilog="""
```
Comment (rcasallas-silabs) [inline scripts/tools/silabs/retrieve_sl_commits.py:46]: I think this is a typo

**Review by rcasallas-silabs — APPROVED** [review]

**Review by lpbeliveau-silabs — APPROVED** [review]
Approve with the condition to look into the typo fix Ricardo pointed out.

File: scripts/tools/silabs/retrieve_sl_commits.py (hunk 1 above)
Comment (mkardous-silabs) [reply scripts/tools/silabs/retrieve_sl_commits.py:46]: @rcasallas-silabs @lpbeliveau-silabs It's not a typo - at least for the restyler.
epilog is an function argument so the restyler alligned it with the function arguemnts

Comment (rcasallas-silabs) [reply scripts/tools/silabs/retrieve_sl_commits.py:46]: Restyler does what restyler does...

**Pull request merged**